src/runtime/hexagon/ops/conv2d.h

```diff
@@ -75,15 +77,31 @@ inline void* to_ptr(uintptr_t v) { return reinterpret_cast<void*>(v); }
 
 inline uintptr_t to_uint(void* ptr) { return reinterpret_cast<uintptr_t>(ptr); }
 
-constexpr int xyc_to_sm_16b(int y, int x, int c) {
+inline constexpr int yxc_to_sm_16b(int y, int x, int c) {
   // Map y,x,c coordinates within a block to the offset (in 16-bit elements)
   // from the beginning of the block in spatial-major layout.
   // 10-bit spatial mask: yyyxcccccx
   assert(y >= 0 && x >= 0 && c >= 0);
   return y << 7 | (x & 2) << 5 | c << 1 | (x & 1);
 }
 
-constexpr int hwio_to_sm_16b(int width, int y, int x, int i, int o) {
+inline constexpr int yxc_to_sm_8b(int y, int x, int c) {
+  // Map y,x,c coordinates within a block to the offset (in 8-bit elements)
+  // from the beginning of the block in spatial-major layout.
+  // 10-bit spatial mask: yyyxxxccccc
```

Comment:
I consciously avoided the checks here because these functions are used for indexing within the innermost loops and need to be really fast. I actually was planning to remove the check from the above `yxc_to_sm_16b` function as well.

I thought I had was to add the `assert` statements and then disable them for release builds with `#define NDEBUG`. Not sure if there's a better solution.

Reply:
I'm pretty sure we can rely on `assert` being disabled when `CMAKE_BUILD_TYPE=Release`.  See https://stackoverflow.com/questions/34302265/does-cmake-build-type-release-imply-dndebug.

Reply:
Another option is to check the loop bounds to make sure y, x and c can't get bigger than can be expressed.